src/main/java/binnie/core/craftgui/minecraft/GuiCraftGUI.java

```diff
@@ -506,4 +523,11 @@ public void drawRect(float x, float y, float width, float height, int color) {
         GL11.glDisable(GL11.GL_BLEND);
         GL11.glEnable(GL11.GL_TEXTURE_2D);
     }
+
+    public ItemStack getStackUnderMouse(int x, int y) {
+        Slot slot = window.calculateMousedOverWidgets().stream().filter(widget -> widget instanceof ControlSlot)
+                .findFirst().map(w -> ((ControlSlot) w).slot).orElse(null);
```

Comment:
You can chain methods like `.map(w -> ((ControlSlot) w).slot).map(Slot::getStack).orElse(null)`